pkg/pki/k8scsrpki/k8scsr_user.go

```diff
@@ -0,0 +1,266 @@
+		if ref.Kind == user.Kind && ref.Name == user.Name {
+			isUserOwnedSecret = true
+			break
+		}
+	}
+	if !isUserOwnedSecret {
+		return nil, errors.New(fmt.Sprintf("secret: %s does not belong to this KafkaUser", secret.Name))
+	}
+	signingRequestGenName, ok := secret.Annotations[DependingCsrAnnotation]
+	if !ok {
+		// Generate new SigningRequest resource
+		signingReq, err = c.generateAndCreateCSR(ctx, secret.Data[corev1.TLSPrivateKeyKey], user)
+		if err != nil {
+			return nil, err
+		}
+
+		if err = c.secretUpdateAnnotation(ctx, secret, signingReq.GetName()); err != nil {
+			return nil, err
+		}
+	}
+	if signingReq == nil {
+		signingReq, err = c.getUserSigningRequest(ctx, signingRequestGenName, secret.GetNamespace())
+		// Handle case when signing request is not found
+		// as like kubernetes removed the signing request
+		if apierrors.IsNotFound(err) {
+			// Generate signing request object and create it
+			if _, ok := secret.Data[corev1.TLSCertKey]; !ok {
+				delete(secret.Annotations, DependingCsrAnnotation)
+				typeMeta := secret.TypeMeta
+				err = c.client.Update(ctx, secret)
+				if err != nil {
+					return nil, err
+				}
+				secret.TypeMeta = typeMeta
+				return nil, errorfactory.New(errorfactory.ResourceNotReady{},
+					errors.New("instance not found"), "kubernetes deleted the csr request",
+					"csrName", signingRequestGenName)
+			}
+		} else if err != nil {
+			return nil, err
+		}
+	}
+	// Handle case when signing request is present
+	var foundApproved bool
+	for _, cond := range signingReq.Status.Conditions {
+		c.logger.Info(fmt.Sprintf("Signing request condition is: %s", cond.Type))
+		if cond.Type == certsigningreqv1.CertificateApproved {
+			foundApproved = true
+			break
+		}
+	}
+
+	if !foundApproved {
+		return nil, errorfactory.New(errorfactory.FatalReconcileError{}, errors.New("instance is not approved"),
+			"could not find approved csr", "csrName", signingReq.GetName())
+	}
+	if len(signingReq.Status.Certificate) == 0 {
+		return nil, errorfactory.New(errorfactory.ResourceNotReady{},
+			errors.New("instance is not ready yet"), "certificate to csr status field is not generated yet",
+			"csrName", signingReq.GetName())
+	}
+
+	certs, err := certutil.ParseCertificates(signingReq.Status.Certificate)
+	if err != nil {
+		return nil, err
+	}
+
+	//Leaf cert
+	secret.Data[corev1.TLSCertKey] = certs[len(certs)-1].ToPEM()
+	//Root CA cert
+	if certs[0].Certificate.IsCA {
+		secret.Data[v1alpha1.CoreCACertKey] = certs[0].ToPEM()
+	}
+
+	certBundleX509 := certutil.GetCertBundle(certs)
+
+	// Ensure a JKS if requested
+	var jks, jksPasswd []byte
+	if user.Spec.IncludeJKS {
+		// we don't have an existing one - make a new one
+		if value, ok := secret.Data[v1alpha1.TLSJKSKeyStore]; !ok || len(value) == 0 {
+			jks, jksPasswd, err = certutil.GenerateJKS(certBundleX509, secret.Data[corev1.TLSPrivateKeyKey])
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	secret.Data[v1alpha1.TLSJKSKeyStore] = jks
```

Comment:
It seems to me when `user.Spec.IncludeJKS` is true and `secret.Data[v1alpha1.TLSJKSKeyStore]` already has a value set, the reconciler overwrites it with an empty jks and jksPasswd string. Is this inteded?

Reply:
Oops nice catch. Thanks.